ganga/GangaDirac/Lib/Server/DiracCommands.py

```diff
@@ -405,3 +405,62 @@ def checkSEStatus(se, access = 'Write'):
     '''
     result = dirac.checkSEAccess(se, access)
     return result
+
+@diracCommand
+def listFiles(baseDir, days):
+
+    from DIRAC.Resources.Catalog.FileCatalog import FileCatalog
+    fc = FileCatalog()
+
+    withMetaData = False
+    if days:
+        withMetaData = True    
+
+    from datetime import datetime, timedelta
+
+    def isOlderThan(cTimeStruct, days):
+        timeDelta = timedelta( days = days )
+        maxCTime = datetime.utcnow() - timeDelta
+        if cTimeStruct < maxCTime:
+            return True
+        return False
+
+    baseDir = baseDir.rstrip('/')
+    print ("Will search for files in %s" % baseDir)
+
+    activeDirs = [baseDir]
+
+    allFiles = []
+    emptyDirs = []
+
+    while len(activeDirs) > 0:
+        currentDir = activeDirs.pop()	
+        res = fc.listDirectory(currentDir, withMetaData, timeout = 360)
+        if not res['OK']:
+            raise Exception("Error retrieving directory contents", "%s %s" % ( currentDir, res['Message'] ) )
```

Comment:
It doesn't make sense to raise exceptions here. If something doesn't work it should be propagated to the `return`. I *think* `execute` should catch if you don't return a dict so simply returning a string with an error message would be fine.

Reply:
Ok, I've changed this now.